lib/workers/global/config/parse/index.ts

```diff
@@ -109,6 +110,16 @@ export async function parseConfigs(
     config.onboardingNoDeps = 'enabled';
   }
 
+  // do not add these secrets to repoSecrets and,
+  //  do not delete the secrets object after applying on global config as it needs to be re-used for repo config
+  if (is.nonEmptyObject(config.secrets)) {
+    config = applySecretsToConfig(config, undefined, false);
+    // adding these secrets to the globalSecrets set so that they can be redacted from logs
+    for (const secret of Object.values(config.secrets!)) {
+      addSecretForSanitizing(secret, 'global');
```

Comment:
Isn't this already done in https://github.com/renovatebot/renovate/blob/ad5c45f43770e8d422c3e93b1c7e805848aa8dc9/lib/config/secrets.ts#L127 ?

Reply:
This adds it to the `repoSecrets` set in the same file, which gets cleared before repo processing begins. That's why I passed the `global` argument